src/greedy.jl

```diff
@@ -133,10 +154,12 @@ function assemble(info::NamedTuple, H::AffineDecomposition, grid, greedy::Greedy
         # Update basis with new snapshot/vector/metric and compute reduced terms
         h_cache = HamiltonianCache(info.h_cache, ext.basis)
 
-        # Update iteration state info
+        # Update iteration state info (use Base.structdiff to remove keys from NamedTuple)
```

Comment:
Hmm. Not sure ... how widespread is the use of this (non-exported) function. Can we rely on the interface to be stable?

Reply:
Apparently this is used in a few packages and there are plans to export it, see https://github.com/JuliaLang/julia/issues/34772. So I would expect that this interface should stay stable, maybe up to a name change.